src/OrbitGl/BatcherImpl.h

```diff
@@ -0,0 +1,102 @@
+// Copyright (c) 2022 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef ORBIT_GL_BATCHER_IMPL_H_
+#define ORBIT_GL_BATCHER_IMPL_H_
+
+#include <Batcher.h>
+
+#include <array>
+#include <memory>
+#include <unordered_map>
+#include <vector>
+
+#include "Containers/BlockChain.h"
+#include "PickingManager.h"
+
+struct LineBuffer {
+  void Reset() {
+    lines_.Reset();
+    colors_.Reset();
+    picking_colors_.Reset();
+  }
+
+  static const int NUM_LINES_PER_BLOCK = 64 * 1024;
+  orbit_containers::BlockChain<Line, NUM_LINES_PER_BLOCK> lines_;
+  orbit_containers::BlockChain<Color, 2 * NUM_LINES_PER_BLOCK> colors_;
+  orbit_containers::BlockChain<Color, 2 * NUM_LINES_PER_BLOCK> picking_colors_;
+};
+
+struct BoxBuffer {
+  void Reset() {
+    boxes_.Reset();
+    colors_.Reset();
+    picking_colors_.Reset();
+  }
+
+  static const int NUM_BOXES_PER_BLOCK = 64 * 1024;
+  orbit_containers::BlockChain<Box, NUM_BOXES_PER_BLOCK> boxes_;
+  orbit_containers::BlockChain<Color, 4 * NUM_BOXES_PER_BLOCK> colors_;
+  orbit_containers::BlockChain<Color, 4 * NUM_BOXES_PER_BLOCK> picking_colors_;
+};
+
+struct TriangleBuffer {
+  void Reset() {
+    triangles_.Reset();
+    colors_.Reset();
+    picking_colors_.Reset();
+  }
+
+  static const int NUM_TRIANGLES_PER_BLOCK = 64 * 1024;
+  orbit_containers::BlockChain<Triangle, NUM_TRIANGLES_PER_BLOCK> triangles_;
+  orbit_containers::BlockChain<Color, 3 * NUM_TRIANGLES_PER_BLOCK> colors_;
+  orbit_containers::BlockChain<Color, 3 * NUM_TRIANGLES_PER_BLOCK> picking_colors_;
+};
+
+struct PrimitiveBuffers {
+  void Reset() {
+    line_buffer.Reset();
+    box_buffer.Reset();
+    triangle_buffer.Reset();
+  }
+
+  LineBuffer line_buffer;
+  BoxBuffer box_buffer;
+  TriangleBuffer triangle_buffer;
+};
```

Comment:
> Since you're already changing the names, would you consider moving Batcher and BatcherInterface into the proper namespace?

Will add it for OpenGlBatcher. After changing Batcher to BatcherInterface I will also include it in orbit_gl namespace.